### PR TITLE
feat: add rpc support and fix recipient amount bug in response

### PR DESCRIPTION
**Pull request by Dushyantgoswami**

### CODE REVIEW

**Comment by axatbhardwaj** [comment]
hi @Dushyantgoswami 

- Great work in Implementing RPC like interface that seems perfect 

- the parsing to float seems a good suggestion but i deliberately tried to keep it in a string format to save it from getting into trouble with large numbers as JS is notorious while using Large numbers 
but if there were some errors using it as a string then you can check BIGINT and fix it using those only 
 Either way we will have to skip float in favor of BIGINT if there was a bug 
would like to request you to recreate the Bug regarding recipient amount and post supporting SS ( i will really appreciate it )

- merging it for now will try to use BIGINT for this

File: blockchainUtils.js [hunk 1]
```diff
@@ -15,16 +15,21 @@ function publicKeyToAddress(publicKey) {
     return ripemd160;
 }
 
-function createTransaction(publicKey, recipient, amount, privateKey) {
+function createTransaction(publicKey, senderAddress, recipient, amount, privateKey) {
     const key = ec.keyFromPrivate(privateKey, 'hex');
+
+    //convert string amount to number
+    amount = parseFloat(amount);
```
Comment (axatbhardwaj) [inline blockchainUtils.js:22]: will use bigINT here

Comment (Dushyantgoswami) [reply blockchainUtils.js:22]: image

this bug occurs when the amount didn't convert to a number, it only affects the recipient balance amount and I don't know how bigInt works, so I have to learn about it first

Comment (Dushyantgoswami) [reply blockchainUtils.js:22]: that's a silly mistake from my side, I passed the amount as a string. sorry for the inconvenience. I created a PR to fix this.
PR #2

**Pull request merged**